docs/config/01-configuration-file.md

```diff
@@ -277,6 +277,15 @@ upon the completion of running the tests. Setting this to false is useful when e
 
 If true, Karma does not display the banner and browser list. Useful when using karma on component tests with screenshots.
 
+## client.allowedReturnUrls
```

Comment:
> Hmm, wouldn't the option in its current state disallow any existing usage of the `return_url` unless the user updates the config? Thus resulting in a breaking change.

I had that concern myself. I thought it might not be under the breaking change (major update) category, because the API is not changing *per se*. Users should only need to change their configuration file. However, I agree that a change even in the configuration is a breaking change.

Reply:
Follow up comments to your other points are coming soon :) I need to be afk.

Reply:
> IMO it makes sense to only allow http/https URLs by default and if the user wants extra security, they can modify the property. So maybe set default as ['^https?://'] and allow users to override it if they want to disallow all URLs or only allow some specific ones. Then we can swap the default to [] in the next major if at all. My thinking is that Open Redirect is not very applicable for Karma anyways - users would not normally click on Karma links.

>Or maybe we should even take a bigger action and get rid of the redirect_url feature altogether. It was added in https://github.com/karma-runner/karma/commit/6af2c897f3b35060a146efdef7da597ba53d8cdd 9 years ago and there is no issue outlining the motivation for this feature and no other mentions of it, so I'm not sure if anybody is using it at all. So maybe removing the feature would be the best mitigation here 😅

I will fix this CL to allow `https?://` by default. That way it will not be a breaking change. And in the next major release we can definitely get rid of `return_url`; I agree with your line of thinking.

Reply:
Done.